### PR TITLE
[scudo] Fix the logic of MaxAllowedFragmentedPages

### PR DESCRIPTION
**Pull request by ChiaHungDuan**
MTE doesn't support MaxReleasedCachePages which may break the assumption that only the first 4 pages will have memory tagged.

### CODE REVIEW

**Comment by ChiaHungDuan** [comment]
I'm running some tests to confirm the issue has been fixed.

**Review by cferris1000 — APPROVED** [review]
LGTM.

File: compiler-rt/lib/scudo/standalone/secondary.h [hunk 1]
```diff
@@ -72,13 +72,16 @@ namespace {
...
-  //   * MaxReleasedCachePages default is currently 4
-  //        - We arrived at this value after noticing that mapping
-  //        in larger memory regions performs better than releasing
-  //        memory and forcing a cache hit. According to the data,
-  //        it suggests that beyond 4 pages, the release execution time is
-  //        longer than the map execution time. In this way, the default
-  //        is dependent on the platform.
+  // We allow a certain amount of fragmentation and part of the fragmented bytes
+  // will be released by `releaseAndZeroPagesToOS()`. This increases the chance
+  // of cache hit rate and reduce the overhead to the RSS at the same time. See
```
Comment (cferris1000) [inline compiler-rt/lib/scudo/standalone/secondary.h:77]: reduces

**Review by cferris1000 — COMMENTED** [review]
Supremely minor nit.

File: compiler-rt/lib/scudo/standalone/secondary.h (hunk 1 above)
Comment (ChiaHungDuan) [reply compiler-rt/lib/scudo/standalone/secondary.h]: Done

**Review by cferris1000 — APPROVED** [review]
LGTM

**Pull request merged**